core/primitives/src/crypto.rs

```diff
@@ -266,17 +268,17 @@ impl<T: AsMut<[u8]> + AsRef<[u8]> + Default + Derive> Ss58Codec for T {
 ///
 /// For now it just specifies how to create a key from a phrase and derivation path.
 #[cfg(feature = "std")]
-pub trait Pair: Sized {
+pub trait Pair: Sized + Send + Sync + 'static {
 	/// TThe type which is used to encode a public key.
-	type Public;
+	type Public: Eq + PartialEq + Clone + Sync + Send + Encode + Decode + Debug + Hash + 'static + AsRef<Self::Public>;
```

Comment:
these extra bounds have broken tests

Reply:
which of them are actually needed?

Reply:
(we should put bounds on the aura function _instead_, it's better practice)

Reply:
> (we should put bounds on the aura function _instead_, it's better practice)

I agree.  The only reason I placed the bounds here is for better diagnostics when developing.

Reply:
Fixed.